nilearn/datasets/func.py

```diff
@@ -1276,3 +1276,189 @@ def fetch_mixed_gambles(n_subjects=1, data_dir=None, url=None, resume=True,
+
+    message = ("The %s you have given '%s' is invalid. Please choose either "
+               "of them %s or list of specific choices.")
+
+    error_correcting_names = ['dimensionality', 'timeseries']
+    user_inputs = [dimensionality, timeseries]
+    assign_names = ['dimensionalities', 'timeseries_methods']
+    standard_variables = [dimensionalities, timeseries_methods]
+
+    for name, check_in, assign, standard in zip(error_correcting_names, user_inputs,
+                                                assign_names, standard_variables):
+        if check_in is not None:
+            if isinstance(check_in, list):
+                for each_str in check_in:
+                    if each_str not in standard:
+                        raise ValueError(message % (
+                            name, check_in, str(standard)))
+                if assign == 'dimensionalities':
+                    dimensionalities = check_in
+                else:
+                    timeseries_methods = check_in
+            elif not isinstance(check_in, list):
+                if check_in not in standard:
+                    raise ValueError(message % (
+                        name, check_in, str(standard)))
+
+                if assign == 'dimensionalities':
+                    dimensionalities = [check_in]
+                else:
+                    timeseries_methods = [check_in]
+
+    n_combinations = len(dimensionalities) * len(timeseries_methods)
+    dataset_name = 'Megatrawls'
+    files_netmats1 = []
+    files_netmats2 = []
+    for dim in dimensionalities:
+        filename = os.path.join('3T_Q1-Q6related468_MSMsulc_d' + str(dim))
+        for timeseries in timeseries_methods:
+            if timeseries == 'multiple_spatial_regression':
+                timeseries = 'ts2'
+            elif timeseries == 'eigen_regression':
+                timeseries = 'ts3'
+            each_files_netmats1 = [(os.path.join(
+                filename + '_' + str(timeseries), 'Znet1.txt'), url, opts)]
+            each_files_netmats2 = [(os.path.join(
+                filename + '_' + str(timeseries), 'Znet2.txt'), url, opts)]
+            files_netmats1.append(each_files_netmats1)
+            files_netmats2.append(each_files_netmats2)
+
+    data_dir = _get_dataset_dir(dataset_name, data_dir=data_dir, verbose=verbose)
+    description = _get_dataset_descr(dataset_name)
+
+    network_matrices1 = []
+    network_matrices2 = []
+    for n in range(n_combinations):
+        netmats1 = _fetch_files(
+            data_dir, files_netmats1[n], resume=resume, verbose=verbose)
+        network_matrices1.extend(netmats1)
+        netmats2 = _fetch_files(
+            data_dir, files_netmats2[n], resume=resume, verbose=verbose)
+        network_matrices2.extend(netmats2)
+
+    return Bunch(Fullcorrelation=network_matrices1,
+                 Partialcorrelation=network_matrices2,
+                 description=description)
```

Comment:
How is the user supposed to know which file corresponds to the timeseries / dimensionality he asked for?